packages/headless/src/utils/builder.ts

```diff
@@ -0,0 +1,43 @@
+import { computed, type ComputedRef, type HtmlHTMLAttributes, type Ref } from "vue";
+
+export type IteratedHeadlessElementFunc<T extends Record<string, unknown>> = (
+  opts: T,
+) => HtmlHTMLAttributes;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type HeadlessElementAttributes = HtmlHTMLAttributes | IteratedHeadlessElementFunc<any>;
+
+export type HeadlessElements = Record<
+  string,
+  HeadlessElementAttributes | ComputedRef<HeadlessElementAttributes>
+>;
```

Comment:
```suggestion
export type HeadlessElements = Record<string, MaybeRef<HeadlessElementAttributes>>;
```

There is a build in vue type for this :)

Reply:
I want a readonly Ref, thats why I use ComputedRef